### PR TITLE
Release/0.21.0

### PR DESCRIPTION
**Pull request by cassava**
Release 0.21.0 PR.

I recommend reviewing the PR commit-by-commit. Especially because some of the "Migrate package to Conan 2.0 compatibility" commits can be a bit much; seeing all the changes at once is overwhelming.

Once you realize the changes are quite simple, but just need to be applied everywhere, the reviewing should be easier.

### CODE REVIEW

File: docs/contributing/creating-a-new-release.rst [hunk 1]
```diff
@@ -55,9 +55,16 @@ First, set the ``VERSION`` file to ``X.Y.Z``::
...
+to use a separate, temporary Conan data directory::
+
+    export CONAN_USER_HOME=~/.var/cloe-X.Y.Z
+    mkdir $CONAN_USER_HOME
+    make setup-conan
+
 Then, compile the entire project locally::
 
-    make purge-all export-vendor export smoketest-deps smoketest
+    make purge-all export smoketest-deps smoketest
```
Comment (clonker) [inline docs/contributing/creating-a-new-release.rst:67]: export-all? or is this only on develop?

File: optional/vtd/CMakeLists.txt [hunk 2]
```diff
@@ -22,13 +23,17 @@ add_library(vtd-object-lib STATIC
...
+    protobuf::libprotobuf
+)
 target_link_libraries(vtd-object-lib
   PUBLIC
     vtd::api
     cloe::runtime
     cloe::models
     Boost::boost
-    open-simulation-interface::open-simulation-interface
+    open_simulation_interface::open_simulation_interface
```
Comment (clonker) [inline optional/vtd/CMakeLists.txt:36]: is OSI required at runtime? can this be a private build dependency?

File: optional/vtd/vendor/osi-sensor-1.0.0/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    def export_sources(self):
+        files.export_conandata_patches(self)
+
+    def requirements(self):
+        self.requires("open-simulation-interface/3.0.1@cloe/stable", private=True)
+        self.requires(
+            "protobuf/2.6.1@cloe/stable",
+            transitive_headers=True,
+            transitive_libs=True,
+            private=True,
```
Comment (clonker) [inline optional/vtd/vendor/osi-sensor-1.0.0/conanfile.py:39]: N :icecream: !

File: optional/vtd/vendor/vtd-2022.3/conanfile.py [hunk 4]
```diff
@@ -9,8 +9,11 @@
 from pathlib import Path
 from typing import List
 
-from conans import CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools import files
+from conans.error import ConanInvalidConfiguration
```
Comment (cassava) [inline optional/vtd/vendor/vtd-2022.3/conanfile.py:14]: Use `conan.error`.

File: optional/vtd/vendor/osi-sensor-1.0.0/patches/cmakelists.patch [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+-else()
+-	set(PRIVATE_LOG_PATH "/tmp/OSMPDummySensorLog.log" CACHE FILEPATH "Path to write private log file to")
++project(OSMPDummySensor LANGUAGES CXX)
++
++# Input Variables
++set(VTD_ROOT "" CACHE STRING "Path to VTD root directory")
++
++# Dependencies
++find_package(open_simulation_interface REQUIRED)
++find_package(protobuf CONFIG REQUIRED)
```
Comment (cassava) [inline optional/vtd/vendor/osi-sensor-1.0.0/patches/cmakelists.patch:24]: Protobuf

File: optional/vtd/vendor/osi-sensor-1.0.0/CMakeLists.txt [hunk 6]
```diff
@@ -1,24 +1,29 @@
...
-# Compiler definitions
-string(TIMESTAMP FMUTIMESTAMP UTC)
-string(MD5 FMUGUID modelDescription.in.xml)
-configure_file(modelDescription.in.xml modelDescription.xml @ONLY)
+# Input Variables
+set(VTD_ROOT "" CACHE STRING "Path to VTD root directory")
+
+# Dependencies
+find_package(open_simulation_interface REQUIRED)
+find_package(protobuf CONFIG REQUIRED)
```
Comment (cassava) [inline optional/vtd/vendor/osi-sensor-1.0.0/CMakeLists.txt:12]: Protobuf

File: docs/contributing/creating-a-new-release.rst (hunk 1 above)
Comment (cassava) [reply docs/contributing/creating-a-new-release.rst:67]: Only on develop.

File: docs/contributing/creating-a-new-release.rst [hunk 7]
```diff
@@ -105,3 +112,44 @@ like ripgrep to search the project for strings containing ``0.19``.
...
+
+E. Create and Merge Pull Request
+--------------------------------
+
+From the branch that we created in step A3, create a new pull request to the
+master branch. If you had a draft pull request, convert it to "ready for review".
+
+In addition to the CI checks, run the Docker builds on your local machine:
+
+
```
Comment (cassava) [inline docs/contributing/creating-a-new-release.rst:124]: Looks like something is missing...

File: docs/contributing/creating-a-new-release.rst [hunk 8]
```diff
@@ -105,3 +112,44 @@ like ripgrep to search the project for strings containing ``0.19``.
...
+Replace ``X.Y.Z`` with the corresponding values.
+
+Then, push the tag to Github::
+
+    git push vX.Y.Z
+
+G. Trigger Read-the-Docs
+------------------------
+
+Login to `ReadTheDocs <https://readthedocs.org>`_ and goto the
```
Comment (cassava) [inline docs/contributing/creating-a-new-release.rst:149]: This is automatic now!

File: optional/vtd/CMakeLists.txt (hunk 2 above)
Comment (cassava) [reply optional/vtd/CMakeLists.txt:36]: I'm not sure. I'd prefer to leave as is for now.

**Comment by cassava** [comment]
Had to rebase to include the latest changes from master.

File: optional/vtd/vendor/protobuf-2.6.1/test_package/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(test_package LANGUAGES CXX)
+
+find_package(protobuf CONFIG REQUIRED)
```
Comment (clonker) [inline optional/vtd/vendor/protobuf-2.6.1/test_package/CMakeLists.txt:5]:
```suggestion
find_package(Protobuf CONFIG REQUIRED)
```

**Review by clonker — CHANGES_REQUESTED** [review]

File: optional/vtd/vendor/protobuf-2.6.1/test_package/CMakeLists.txt (hunk 9 above)
Comment (cassava) [reply optional/vtd/vendor/protobuf-2.6.1/test_package/CMakeLists.txt:5]: Good catch.

**Review by clonker — APPROVED** [review]

**Pull request merged**